test/unified_memory_allocation/uma_pools/disjoint_pool_config.cpp

```diff
@@ -0,0 +1,325 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: MIT
+
+#include "helpers.hpp"
+#include "usm_pool_config.hpp"
+
+using PoolDescriptor = usm_pool::PoolDescriptor;
+
+const PoolDescriptor DESCRIPTORS[4] = {PoolDescriptor(UR_USM_TYPE_HOST),
+                                       PoolDescriptor(UR_USM_TYPE_DEVICE),
+                                       PoolDescriptor(UR_USM_TYPE_SHARED),
+                                       PoolDescriptor(UR_USM_TYPE_SHARED, true)};
+
+TEST(disjointPoolConfigTests, disjointPoolConfigTest) {
+    // test if the returned unordered map contains only the valid configs
+    usm_pool::PoolConfigurations<typename DisjointPool::Config> configsMap =
+        usm_pool::getPoolConfigurationsFor<DisjointPool>();
+
+    // check if all the valid configs are present
+    for (const auto &descriptor : DESCRIPTORS) {
+        ASSERT_EQ(configsMap.count(descriptor), 1);
+    }
+
+    // check if there are no more configs in the unordered map
+    ASSERT_EQ(configsMap.size(), 4);
+}
+
+TEST(disjointPoolConfigTests, disjointPoolConfigStringEnabledBuffersTest) {
+    // test for valid string with enabled buffers-- (values to configs should be
+    // parsed from string)
+    std::string config = "1;32M;host:1M,4,64k;device:1m,4,64K;shared:0,3,1M;"
+                         "read_only_shared:0,0,3M";
+    usm_pool::PoolConfigurations<typename DisjointPool::Config> configsMap =
+        usm_pool::getPoolConfigurationsFor<DisjointPool>(config);
+
+    // test for host
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[0]).MaxPoolableSize, 1 * 1024 * 1024);
```

Comment:
You can make a test class with all default values defined. Easier to spot whether you check against a default value or one defined with a config string.

Reply:
Done.